### PR TITLE
RavenDB-18224: add more details & fix NRE in GetSubscriptionResend endpoint

### PR DESCRIPTION
**Pull request by garayx**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18224

### Additional description

- add more details to `GetSubscriptionResend` endpoint
- fix NRE in GetSubscriptionResend endpoint (`subscriptionConnections.GetActiveBatches()` can be null if there is no connection)

### Type of change

- Bug fix

### How risky is the change?

- Not relevant

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing by Contributor

- It has been verified by manual testing

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs [hunk 1]
```diff
@@ -231,29 +232,97 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
         [RavenAction("/databases/*/debug/subscriptions/resend", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
         public async Task GetSubscriptionResend()
         {
-            var subscriptionName = GetStringQueryString("name");
+            var subscriptionName = GetStringQueryString("name", required: false);
+            var detailed = GetBoolValueQueryString("detailed", required: false);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]:
```
GetBoolValueQueryString("detailed", required: false) ?? false
```

File: src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs [hunk 2]
```diff
@@ -231,29 +232,97 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
...
+                    first = false;
+
+                    var name = Database.SubscriptionStorage.GetSubscriptionNameById(context, kvp.Key);
+                    var subscriptionConnections = name == null ? null : Database.SubscriptionStorage.GetSubscriptionConnectionsState(context, subscriptionName);
+
+                    List<DynamicJsonValue> itemsJson;
+                    if (detailed is true)
+                    {
+                        using (ContextPool.AllocateOperationContext(out DocumentsOperationContext documentsContext))
+                        using (documentsContext.OpenReadTransaction())
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs]: let's have a state from one point in time. Open that tx before the loop?

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs [hunk 3]
```diff
@@ -430,22 +431,42 @@ public DynamicJsonValue ToJson()
...
+                    };
+                }
+            }
+        }
 
-                    yield return resendItem;
+        public static IEnumerable<ResendItem> GetResendItemsForDatabase(ClusterOperationContext context, string database)
+        {
+            var subscriptionState = context.Transaction.InnerTransaction.OpenTable(ClusterStateMachine.SubscriptionStateSchema, ClusterStateMachine.SubscriptionState);
+            using var _ = Slice.From(context.Allocator, database.ToLowerInvariant(), out var dbName);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs]: missing terminator?

Comment (garayx) [reply src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs]: yes done

File: src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs [hunk 4]
```diff
@@ -430,22 +433,42 @@ public DynamicJsonValue ToJson()
...
+                    };
+                }
+            }
+        }
 
-                    yield return resendItem;
+        public static IEnumerable<ResendItem> GetResendItemsForDatabase(ClusterOperationContext context, string database)
+        {
+            var subscriptionState = context.Transaction.InnerTransaction.OpenTable(ClusterStateMachine.SubscriptionStateSchema, ClusterStateMachine.SubscriptionState);
+            using var _ = Slice.From(context.Allocator, database.ToLowerInvariant(), SpecialChars.RecordSeparator, ByteStringType.Immutable, out var dbNamePrefix);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs:459]: are we putting items using RecordSeparator (30) or `/` (47) ?

Comment (garayx) [reply src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs:459]: I used the same one as in other subscriptions methods:
https://github.com/garayx/ravendb/blob/1ed42da34be3ed8c75a32581f86db9f6951dff95/src/Raven.Server/Documents/Subscriptions/SubscriptionConnectionsState.cs#L567

Comment (ppekrol) [reply src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs:459]: Awesome, wanted to be sure

**Review by ppekrol — APPROVED** [review]

**Comment by ppekrol** [comment]
Please check failing tests

**Comment by ppekrol** [comment]
Please fix the warnings

**Pull request merged**